### PR TITLE
Adding explanations to default guardrails

### PR DESCRIPTION
**Pull request by roh26it**
**Adding explanations to the data property of default guardrail plugins** 
We're adding a `data.explanation` property to default guardrails to make it easier to identify why they succeeded or failed.

### CODE REVIEW

**Comment by VisargD** [comment]
Changing this to draft. Please update the PR once its ready

**Comment by roh26it** [comment]
@VisargD all tests for default plugins are passing. You'll need to check the portkey plugins with credentials. I could not run those tests

File: package.json [hunk 1]
```diff
@@ -46,6 +46,8 @@
     "@portkey-ai/mustache": "^2.1.2",
     "@smithy/signature-v4": "^2.1.1",
     "@types/mustache": "^4.2.5",
+    "ajv": "^8.17.1",
```
Comment (VisargD) [inline package.json]: From what I remember, `ajv` does not support edge runtimes. Let me confirm once.

Comment (VisargD) [reply package.json]: Also, any reason to replace zod with ajv?

Comment (roh26it) [reply package.json]: Replaced it with worker compatible json schema validator now

**Review by VisargD — APPROVED** [review]

**Pull request merged**